### ISSUE TITLE
add_metric() does not work on this dataset

### ISSUE THREAD

**Issue by Scoodood** [pycaret/pycaret#1063]
First of all, I am using 
```
python 3.8.5
pycaret 2.2.3
scikit-learn 0.23.2
Win10 Pro
```

The `add_metric()` works fine on the built-in dataset as shown in below.
```
from pycaret.datasets import get_data
from pycaret.classification import *
from sklearn.metrics import average_precision_score

data1 = get_data('juice')
clf = setup(data1, target='Purchase', session_id=123, silent=True, verbose=False)
add_metric('auprc', 'AUPRC', average_precision_score, target = 'pred_proba')
best = compare_models(include=['rf', 'catboost', 'xgboost', 'lightgbm'], fold=3)
```
![Capture](https://user-images.githubusercontent.com/15989924/107740131-3b0a0d80-6cbf-11eb-8369-669f2869811e.PNG)

However, when I use my own dataset, it stops working for some reason. My output is simply blank. Here is my code.

```python
import pandas as pd
import numpy as np

data2 = pd.DataFrame(np.random.randint(0, 2, size=(1000,20)))
data2.columns = [f'f{ea:02d}' for ea in range(data2.shape[1])]
data2['target'] = np.random.choice(['A', 'B', 'C'], data2.shape[0])
data2.head()
```
![Capture](https://user-images.githubusercontent.com/15989924/107740753-b5875d00-6cc0-11eb-9e53-1d675ddff8ad.PNG)
### ... the following code simply return `blank`
```python
clf = setup(data2, target='target', session_id=123, silent=True, verbose=False)
add_metric('auprc', 'AUPRC', average_precision_score, target = 'pred_proba')
best = compare_models(include=['rf', 'catboost', 'xgboost', 'lightgbm'], fold=3)
```

<hr>

But then I had decided to try another scikit built-in function `log_loss` on the same dataset, it works fine. 

```python
from sklearn.metrics import log_loss
clf = setup(data2, target='target', session_id=123, silent=True, verbose=False)
add_metric('logloss', 'LogLoss', log_loss, target = 'pred_proba')
best = compare_models(include=['rf', 'catboost', 'xgboost', 'lightgbm'], fold=3)
```
![Capture](https://user-images.githubusercontent.com/15989924/107740359-c84d6200-6cbf-11eb-941f-ec7bd86a9229.PNG)

Can someone please shed some light on my `AUPRC`? Thanks

**Comment by Yard1** [comment]
Whenever `compare_models()` returns a blank table, it's best to run it again with `errors="raise"` argument. By doing that, you will receive an exception. In this case, the metric you are using does not support multilabel problems. It is a sklearn issue and will happen without PyCaret.

**Comment by Scoodood** [comment]
You are right, I had confirmed your observation that it was indeed `sklearn.metrics.average_precision_score` issue. The following code raises the same `multiclass` error message on `scikit-learn version 0.23.2`

```
from sklearn.model_selection import cross_validate
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier()
y = data2.target.values
x = data2.drop('target', axis=1)
cross_validate(rf, x, y, scoring='average_precision')
```

I can see the `scoring='roc_auc'` also produce the same error messages. 
```
rf = RandomForestClassifier()
y = data2.target.values
x = data2.drop('target', axis=1)
cross_validate(rf, x, y, scoring='roc_auc')
```

But then since the `roc_auc()` can handle multiclass target with `multiclass='ovr'`, we can fix the code with `scoring='roc_auc_ovr'
```
rf = RandomForestClassifier()
y = data2.target.values
x = data2.drop('target', axis=1)
cross_validate(rf, x, y, scoring='roc_auc_ovr')
```

I can see that [scikit learn](https://github.com/scikit-learn/scikit-learn/pull/17388) have been trying to add the `multiclass` support to `average_precision_score` since May 2020...

**Comment by Yard1** [comment]
Indeed. I'll close this issue as it is not related to PyCaret.

**Comment by Scoodood** [comment]
Thanks @Yard1
